pkg/features/antrea_features.go

```diff
@@ -70,8 +70,30 @@ var (
 		FlowExporter:       {Default: false, PreRelease: featuregate.Alpha},
 		NetworkPolicyStats: {Default: false, PreRelease: featuregate.Alpha},
 	}
+
+	// UnsupportedFeaturesOnWindows records the features not supported on
+	// a Windows Node. Antrea Agent on a Windows Node checks the enabled
+	// features, and fails the startup if an unsupported feature is enabled.
+	// We do not define a separate defaultAntreaFeatureGates map for
+	// Windows, because Agent code assumes all features are registered (
+	// FeatureGate.Enabled(feature) will panic if the feature is not added
+	// to the FeatureGate).
+	// In future, if a feature is supported on both Linux and Windows, but
+	// can have different FeatureSpecs between Linux and Windows, we should
+	// still define a separate defaultAntreaFeatureGates map for Windows.
+	unsupportedFeaturesOnWindows = map[featuregate.Feature]struct{}{}
```

Comment:
shouldn't this be generalized to verify to features that *must* be enabled on Windows (at the moment, I think it's just AntreaProxy?) ?

we could have something like:
```go
windowsFeaturesRequirements = map[featuregate.Feature]bool{
    AntreaProxy: true,
    SomeFeatureNotSupportedOnWindows: false,
}
```
if a feature has no requirements (can be enabled or disabled) it is not added to the map

I actually don't remember if we already have some code to verify that AntreaProxy is enabled on Windows, but even if we do, it could be good to unify with this

Reply:
I think AntreaProxy is not a must for Windows?

Reply:
I thought that the K8s NP implementation was broken on Windows for Services, which is why we enable it by default on Windows? At least, that's what our documentation says. If this is accurate, I think it's best to consider it a must and add this check.

Reply:
That is true. But the current Agent code still assumes Antrea Proxy can be disabled for Windows. How about adding a warning message?